### PR TITLE
CPM-729: Add phpCD for identifier generator components

### PR DESCRIPTION
**Pull request by antoinearrive**
**Description (for Contributor and Core Developer)**

Add phpCD for identifier generator components

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: components/identifier-generator/back/.php_cd.php [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+$rules = [
+    // Domain layer should only use classes from itself and Assert
+    $builder->only(
+        [
+            'Akeneo\Pim\Automation\IdentifierGenerator\Domain',
+            'Webmozart\Assert\Assert'
+        ]
+    )->in('Akeneo\Pim\Automation\IdentifierGenerator\Domain'),
+
+    // Infra layer should only use classes from Domain and Symfony
```
Comment (mmetayer) [inline components/identifier-generator/back/.php_cd.php:22]: this comment will soon be out of date :)

File: components/identifier-generator/back/.php_cd.php [hunk 2]
```diff
@@ -0,0 +1,38 @@
+<?php
+
+declare(strict_types=1);
+
```
[the remark this answers is not in the document]
Comment (mmetayer) [reply components/identifier-generator/back/.php_cd.php:4]: (and move the file in the `tests` directory please)

File: make-file/identifier-generator.mk [hunk 3]
```diff
@@ -9,16 +11,22 @@ identifier-generator-front-fix:
...
 .PHONY: identifier-generator-acceptance-back
 identifier-generator-acceptance-back:
-	$(PHP_RUN) vendor/bin/behat --config components/identifier-generator/back/tests/behat.yml --suite=acceptance --format pim --out var/tests/behat/identifier-generator --format progress --out std --colors $(O)
+	$(PHP_RUN) vendor/bin/behat --config $(IDENTIFIER_GENERATOR_PATH)/back/tests/behat.yml --suite=acceptance --format pim --out var/tests/behat/identifier-generator --format progress --out std --colors $(O)
+
+.PHONY: identifier-generator-coupling-back
+identifier-generator-coupling-back:
+	$(PHP_RUN) vendor/bin/php-coupling-detector detect \
+		--config-file=$(IDENTIFIER_GENERATOR_PATH)/back/.php_cd.php \
+		$(IDENTIFIER_GENERATOR_PATH)/back/src
```
Comment (mmetayer) [inline make-file/identifier-generator.mk:32]:
```suggestion
		--config-file=$(IDENTIFIER_GENERATOR_PATH)/back/tests/.php_cd.php
```

File: components/identifier-generator/back/.php_cd.php (hunk 1 above)
Comment (antoinearrive) [reply components/identifier-generator/back/.php_cd.php:30]: i remove this comment

**Review by mmetayer — APPROVED** [review]

**Review by fchoquet — COMMENTED** [review]
Just a small comment, the rest looks good!

File: components/identifier-generator/back/src/Domain/Repository/IdentifierGeneratorRepository.php [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+use Akeneo\Pim\Automation\IdentifierGenerator\Domain\Model\IdentifierGenerator;
+
+/**
+ * @copyright 2022 Akeneo SAS (https://www.akeneo.com)
+ * @license   https://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+interface IdentifierGeneratorRepository
+{
+    public function save(IdentifierGenerator $identifierGenerator): void;
```
[the remark this answers is not in the document]
Comment (fchoquet) [reply components/identifier-generator/back/src/Domain/Repository/IdentifierGeneratorRepository.php]: reported in the parent PR. Closing here

**Pull request merged**